gmm_client/src/client.erl

```diff
@@ -0,0 +1,105 @@
+%%%-------------------------------------------------------------------
+%%% @author Svatopluk
+%%% @copyright (C) 2021, <COMPANY>
+%%% @doc
+%%%
+%%% @end
+%%% Created : 10. maj 2021 16:54
+%%%-------------------------------------------------------------------
+-module(client).
+-author("Svatopluk").
+
+
+%% API
+-export([get_id_list/0, get_id_number/0, get_user_info/1, delete_user/1, add_user/1, test/0]).
+
+test()->
+  ok.
+
+url_address() ->
+  "localhost:8080/".
+
+% 
+
+get_request(URL)->
+  Method = get,
+  Headers = [],
+  Payload = <<>>,
+  Options = [],
+  {ok, StatusCode, RepsHeaders, ClientRef} = hackney:request(Method, URL, Headers, Payload, Options),
+  {ok, StatusCode, RepsHeaders, ClientRef}.
+
+% reads body of get request
+read_body(ClientRef) ->
+  {ok, _} = hackney:body(ClientRef).
+
+
+% Get Id List Section
+
+get_id_list() ->
+  application:ensure_all_started(hackney),
+  URL = url_address()++"users",
+  {ok, _, _, ClientRef} = get_request(URL),
+  {ok, Body} = read_body(ClientRef),
+  list_body_elements(binary:bin_to_list(Body)).
+
+get_id_number()->
+  application:ensure_all_started(hackney),
+  URL = url_address()++"users",
+  {ok, _, _, ClientRef} = get_request(URL),
+  {ok, Body} = read_body(ClientRef),
+  count_body_elements(binary:bin_to_list(Body), 0).
+
+count_body_elements(Body, Acc) ->
+  case Body of
+    [] -> Acc;
+    Body ->
+      [_ | Tail] = string:split(Body, ","),
+      count_body_elements(Tail, Acc + 1)
+  end.
+
+% Get User Info Section
+
+get_user_info(Id)->
+  application:ensure_all_started(hackney),
+  URL = url_address()++"users/"++Id,
+  {ok, _, _, ClientRef} = get_request(URL),
+  {ok, Body} = read_body(ClientRef),
+  binary:bin_to_list(Body).
```

Comment:
Cały ten kawałeczek kodu w zasadzie powtarzasz w `get_id_list/0`, `get_id_number/1` i tutaj. Brzmi mi to jak kandydat na wyciągnięcie do osobnej funkcji i uproszczenie kodu